addons/html_editor/static/src/utils/dom_info.js

```diff
@@ -731,6 +739,32 @@ export function areSimilarElements(node, node2) {
     );
 }
 
+export function hasSameStyleAttributes(node, node2) {
+    const nodeStyles = node
+        .getAttribute("style")
+        .split(";")
+        .map((style) => style.trim())
+        .filter(Boolean);
+    const node2Styles = node2
+        .getAttribute("style")
+        .split(";")
+        .map((style) => style.trim())
+        .filter(Boolean);
+    return (
+        nodeStyles.length === node2Styles.length &&
+        nodeStyles.every((style) => node2Styles.includes(style))
+    );
+}
+
+export function hasSameClasses(node, node2) {
+    const nodeClasses = node.getAttribute("class").trim().split(/\s+/);
+    const node2Classes = node2.getAttribute("class").trim().split(/\s+/);
```

Comment:
Maybe we can do the same here, even if it's less critical since it's only one line.
So, it's up to you.